### ISSUE TITLE
Frequent Freezes in 20.10 Oryx Pro 7 (2021)

### ISSUE THREAD

**Issue by lvaickus** [pop-os/pop#1640]
**Distribution (run `cat /etc/os-release`):**
NAME="Pop!_OS"
VERSION="20.10"
ID=pop
ID_LIKE="ubuntu debian"
PRETTY_NAME="Pop!_OS 20.10"
VERSION_ID="20.10"
HOME_URL="https://pop.system76.com"
SUPPORT_URL="https://support.system76.com"
BUG_REPORT_URL="https://github.com/pop-os/pop/issues"
PRIVACY_POLICY_URL="https://system76.com/privacy"
VERSION_CODENAME=groovy
UBUNTU_CODENAME=groovy
LOGO=distributor-logo-pop-os

**Hardware:**
oryx pro 7 (2021), intel 10870H, 64GB RAM, 2X 2TB SSD, RTX3080.

**Issue/Bug Description:**
Hard freezes at any point when laptop is powered on with unresponsive mouse and keyboard requiring hard power cycle. In the last 3 days I have had 16 total freezes in maybe 5 hours of up time. 6 freezes after booting to desktop after variable amounts of time ranging from 15 minutes to 4 hours on various tasks: web browsing, working in terminal, etc, nothing computationally or GPU intensive.  I have also had freezes at the encryption password screen after password entered and displayed the "successful" message and at the login screen after entering password. 

One of the crashes corrupted all conda environments even though no conda process was running. 

Perhaps unrelated, but can't be sure: keyboard is unresponsive at encryption screen e.g. I have to hit each key 2-4 times before it's registered. If I wait 2 minutes, the keyboard performs as expected.

I'm in "compute" graphics setting (which in and of itself was a whole saga requiring purging and reinstallation of a bunch of drivers)

I've probably had 50 computers over the last 30 years, many hand built and scavenged from dumpsters and this is the most unstable by a factor of 10. I even had a rig with a 100% overclock and no extra voltage that didn't crash this much.

**Steps to reproduce (if you know):**
Unknown. The only situation where it has occurred more than once is at encryption and log in screen.

**Expected behavior:**
Rare freezes due to extreme situations / stupid stuff I did.

**Comment by romen** [comment]
Are you using BTRFS as your root partition?
If so, this might be relevant: https://github.com/pop-os/default-settings/issues/111

I was experiencing simlar issues using 20.04 as my installed system, but verified they were present also when running the 20.10 live.

The main difference for me was that the freezes would last up to some minutes, but eventually I could regain control just by waiting: it was the worst when many processes were doing many transactions, but given I was trying to determine the cause of the issue I was quite mindful of what kind of processes were being run. I can imagine that with many browser tabs opened and more heavy disk loads the freeze could last seemingly indefinitely.

**Comment by lvaickus** [comment]
Thanks for the input, but it looks like both drives are ext4. 

So I've tried waiting up to 30 minutes and control never comes back (at least in that time frame).

I haven't done anything computationally, GPU or RAM intensive yet. For example in one freeze I had just logged in, opened firefox and navigated to Gmail when it froze.

Another time the only application running was terminal, and I had just hit enter on nvidia-smi and it froze.

I had to power cycle the laptop 4 times to respond to this message. Two freezes immediately after landing on the desktop but before opening any apps, two freezes right after entering login password and hitting enter.

It almost smells like a hardware problem to me, e.g. improperly seated RAM, but the system is brand new and I have heard great things about how extensively system76 tests their machines before shipping. I've also seen posts from other people describing similar behavior on reddit though not on the same hardware.

The only other clue is that after one freeze, nvidia-smi no longer worked and threw this error message: "Unable to determine the device handle for GPU". It reverted to normal behavior after a reboot.

I've got an extensive ticket going with system76, and will keep this thread updated with whatever solution they come up with, and considering how much this laptop cost, it's going to have to be a perfect solution.

Thanks!

**Comment by romen** [comment]
I really hoped my fix could be a solution to your problem too! Best of luck!

**Comment by lvaickus** [comment]
I thought I'd post my resolution here:

So, I continued to have multiple crashes a day. Tech support was friendly, but not responsive enough (e.g. 1 suggestion every 24 hours) and I just did not have the time to continue to bang my head against the wall.

Even though I'm going to eat the shipping cost, I cannot tolerate having a 4000$ laptop that I do not trust implicitly. First impressions being what they are, I was not willing to let them attempt to fix the device at System76 HQ as no computer I've ever bought (including from off brand 1990's builders like Quantex) has ever been this unstable at any point in it's life cycle. I even have a Compaq laptop from 2001 that is still reliable to this day with only a few battery changes.

Perhaps the dream of a Mac-level pure linux laptop with a discrete GPU is still fanciful...Going to wait for the M1X/M2 macbook instead...
